network/peers/conn_manager.go

```diff
@@ -2,34 +2,37 @@ package peers
 
 import (
 	"context"
-	"encoding/base64"
-	"fmt"
-	"math"
-	"sort"
+	"time"
 
+	"github.com/jellydator/ttlcache/v3"
 	connmgrcore "github.com/libp2p/go-libp2p/core/connmgr"
 	libp2pnetwork "github.com/libp2p/go-libp2p/core/network"
 	"github.com/libp2p/go-libp2p/core/peer"
-	"go.uber.org/zap"
-
 	"github.com/ssvlabs/ssv/logging/fields"
 	"github.com/ssvlabs/ssv/network/records"
+	"go.uber.org/zap"
 )
 
 const (
-	protectedTag = "ssv/subnets"
+	ProtectedTag = "ssv/subnets"
 )
 
 type PeerScore float64
 
+var (
+	TrimmedRecently = ttlcache.New(ttlcache.WithTTL[peer.ID, struct{}](30 * time.Minute))
+)
+
+func init() {
+	go TrimmedRecently.Start() // start cleanup go-routine
+}
```

Comment:
I know global vars aren't best way to do it, but I think we should leave it global for now and refactor it later (once this code has been running in production for a while and we've proven this solution really does help). Since we might want/have to swap it out for something else.